scalanet/src/io/iohk/scalanet/peergroup/UDPPeerGroup.scala

```diff
@@ -185,11 +203,17 @@ class UDPPeerGroup[M](val config: Config, cleanupScheduler: Scheduler = Schedule
                 // taking it from map but before server pushes message on to it, `onNext` would be called after `onComplete`
                 // which is breach of observer contract.
                 // It is worth investigating if it is only theoretical possibility or additional synchronization is needed
-                messageE.foreach(m => channel.messageSubject.onNext(m))
+                handleIncomingMessage(channel, datagram)
+              } catch {
+                case NonFatal(e) => handleError(ctx, e)
```

Comment:
Why are we handling only NonFatal here? And isn't `exceptionCaught` called for this errors as well?

Reply:
imo it makes no sense to catch fatal exception here and passing it further as stream event, the proccess will most probably die anyway. 
Tbh i am not what is netty handler policy in case of such errors.